### PR TITLE
Support for persist_docs

### PR DESCRIPTION
**Pull request by hovaesco**
## Overview

- Description: Support for persist_docs
- Related issue(s): https://github.com/starburstdata/dbt-trino/issues/76
- Related code pull request(s): https://github.com/starburstdata/dbt-trino/pull/70 is needed to test the functionality since memory connector doesn't support comments. 
- Related link(s):

## Checklist

- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] `README.md` updated and added information about my change
- [ ] `CHANGELOG.md` updated and added information about my change

### CODE REVIEW

File: tests/functional/adapter/persist_docs/test_persist_docs.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+# https://github.com/trinodb/trino/issues/10705
+class TestPersistDocsView(TestPersistDocsBase):
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "name": "persist_docs_tests",
+            "models": {
+                "+persist_docs": {
+                    "relation": True
```
Comment (findinpath) [inline tests/functional/adapter/persist_docs/test_persist_docs.py:81]: we basically test whether dbt is persisting docs for relations and columns, but can we check whether the comments were set as expected on the underlying database?

File: dbt/adapters/trino/connections.py [hunk 2]
```diff
@@ -442,11 +442,16 @@ def add_query(self, sql, auto_begin=True,
...
+            if conn is None or conn.name is None:
+                conn_name = "<None>"
+            else:
+                conn_name = conn.name
             raise dbt.exceptions.RuntimeException(
                 "Tried to run an empty query on model '{}'. If you are "
                 "conditionally running\nsql, eg. in a model hook, make "
                 "sure your `else` clause contains valid sql!\n\n"
-                "Provided SQL:\n{}".format(connection.name, sql)
+                "Provided SQL:\n{}".format(conn_name, sql)
```
[the remark this answers is not in the document]
Comment (hovaesco) [reply dbt/adapters/trino/connections.py:454]: I've done a mistake in macro and error messages from dbt was smothering like `NoneType` has no attribute `name`.

File: dbt/include/trino/macros/adapters.sql [hunk 3]
```diff
@@ -158,6 +158,11 @@
 {% endmacro %}
 
 
+{% macro trino__alter_relation_comment(relation, relation_comment) -%}
+  comment on {{ relation.type }} {{ relation }} IS '{{ relation_comment | replace("'", "''") }}';
+{% endmacro %}
+
```
[the remark this answers is not in the document]
Comment (hovaesco) [reply dbt/include/trino/macros/adapters.sql:173]: Two new lines are present after each macro in this file.

File: dbt/include/trino/macros/adapters.sql [hunk 4]
```diff
@@ -158,6 +158,20 @@
...
 
 
+{% macro trino__alter_relation_comment(relation, relation_comment) -%}
+  comment on {{ relation.type }} {{ relation }} is '{{ relation_comment | replace("'", "''") }}';
+{% endmacro %}
+
+
+{% macro trino__alter_column_comment(relation, column_dict) %}
+  {% set existing_columns = adapter.get_columns_in_relation(relation) | map(attribute="name") | list %}
+  {% for column_name in column_dict if (column_name in existing_columns) %}
```
[the remark this answers is not in the document]
Comment (hovaesco) [reply dbt/include/trino/macros/adapters.sql:168]: Comments are deleted if there are no set anymore. It's done by setting an empty comment

File: dbt/adapters/trino/connections.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdesmet) [reply dbt/adapters/trino/connections.py:454]: I would say if we don't know exactly know how to reproduce this issue, remove it from the PR.

[the remark this answers is not in the document]
Comment (hovaesco) [reply dbt/adapters/trino/connections.py]: I dropped this commit from PR

File: dbt/include/trino/macros/adapters.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (hovaesco) [reply dbt/include/trino/macros/adapters.sql:168]: Thanks, the code adjusted accordingly to the comment.

File: tests/functional/adapter/persist_docs/test_persist_docs.py (hunk 1 above)
Comment (hovaesco) [reply tests/functional/adapter/persist_docs/test_persist_docs.py]: Thanks for a valid comment. This test should be handled by macro https://github.com/starburstdata/dbt-trino/blob/master/dbt/include/trino/macros/catalog.sql which creates a file `target/catalog.json` generated by `dbt docs generate` command. 

In the `get_catalog` macro table and column comments are returned as `null`. Information schema tables in Trino miss comment columns. I will create a follow-up ticket in Trino for it.  

Docs are generated by a file located in `target/manifest.json` and all comments are present in dbt-docs and persisted in the Trino.

Comment (hovaesco) [reply tests/functional/adapter/persist_docs/test_persist_docs.py]: https://github.com/trinodb/trino/issues/13705

**Review by mdesmet — APPROVED** [review]
LGTM

Maybe add a TODO in the code in the metadata query for the comments with a link to the Trino issue.

**Pull request merged**